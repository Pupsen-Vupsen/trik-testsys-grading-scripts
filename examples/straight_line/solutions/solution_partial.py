import sys
import time
import random
import math

class Program():
  __interpretation_started_timestamp__ = time.time() * 1000

  pi = 3.141592653589793
  x = None

  def execMain(self):

    
    self.x = 0
    brick.motor("M3").setPower(100)
    brick.motor("M4").setPower(100)
    
    script.wait(1250)
    
    while True:
      if brick.sensor("A1").read() < 10:
        break
      if brick.sensor("A1").read() < 40:
        self.x = self.x + 1
        if self.x < 3:
          script.wait(500)
          
          brick.motor("M1").brake()
          brick.motor("M2").brake()
          brick.motor("M3").brake()
          brick.motor("M4").brake()
          
          script.wait(1000)
          
          brick.motor("M3").setPower(100)
          brick.motor("M4").setPower(100)
          
          script.wait(500)
          
    script.wait(500)
    
    brick.motor("M1").brake()
    brick.motor("M2").brake()
    brick.motor("M3").brake()
    brick.motor("M4").brake()
    
    brick.stop()
    return

def main():
  program = Program()
  program.execMain()

if __name__ == '__main__':
  main()
