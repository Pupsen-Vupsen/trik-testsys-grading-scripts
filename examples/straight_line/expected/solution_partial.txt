-------- Results for ./results/straight_line/solution_partial --------
./results/straight_line/solution_partial/1_square.json: 40
./results/straight_line/solution_partial/2_square.json: 40
TOTAL: 80
